streaming/base/util.py

```diff
@@ -217,26 +220,26 @@ def get_import_exception_message(package_name: str, extra_deps: str) -> str:
 
 
 def merge_index(*args: Any, **kwargs: Any):
-    r"""Merge index.json from partitions to form a global index.json.
+    r"""Merge index.json from streams to form a global index.json.
 
     This can be called as
 
         merge_index(index_file_urls, out, keep_local, download_timeout)
 
         merge_index(out, keep_local, download_timeout)
 
-    The first signature takes in a list of index files URLs of MDS partitions.
-    The second takes the root of a MDS dataset and parse the partition folders from there.
+    The first signature takes in a list of index files URLs of MDS streams.
+    The second takes the root of a MDS dataset and parse the streams folders from there.
 
     Args:
-        index_file_urls (List[Union[str, Tuple[str,str]]]): index.json from all the partitions.
+        index_file_urls (List[Union[str, Tuple[str,str]]]): index.json from all the streams.
```

Comment:
This is an entry function and it branches to either merge_index from list or merge_index from a root folder. Are you recommending to not remove this entry function, and just keep the two actual implementations? Can you elaborate a bit?

Reply:
Oh I mean that the signature of the `merge_index` function isn't typed -- it just has `*args, **kwargs`, but the docstring has types for the arguments, for example, `index_file_urls` has type `(List[Union[str, Tuple[str,str]]])`. I'm suggesting we add types to the `merge_index` function so that users will easily be able to see what the `merge_index` function takes in (for example, IDEs all support this, would be better for our docs, etc). Otherwise they have to reference the docstring, which doesn't match up with the function signature. lmk if i should elaborate more

Reply:
Keeping this entry function is fine